org.eclipse.dawnsci.remotedataset.client/src/org/eclipse/dawnsci/remotedataset/client/RemoteLoader.java

```diff
@@ -50,4 +58,40 @@ public IDataset getDataset(IMonitor unused, SliceND slice) throws IOException {
 		return ret;
 	}
 
+	@Override
+	public int[] refreshShape() {
+
+		try {
+			RemoteDatasetInfo datasetInfo = getDatasetInfo();
+			return datasetInfo.getShape();
+		} catch (Exception e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+			return null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Return an empty array instead of null. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1168)